contents/tutorials/array-filter-breakdown.md

```diff
@@ -0,0 +1,88 @@
+
+The most useful breakdown with arrays is `arrayJoin`. This is a [special expression](https://clickhouse.com/docs/en/sql-reference/functions/array-join) that "unfolds" an array into multiple rows. and helps us access each value from within an array. For example, to get a count of the usage of different `$active_feature_flags`, you can use this HogQL expression breakdown:
+
+```sql
+arrayJoin(
+  JSONExtractArrayRaw(
+    properties.$active_feature_flags ?? '[]'
+  )
+)
+```
+
+> Adding `?? '[]'`  is a "nullish coalescing operator" that provides a default value to your properties. In this case, it provides an empty array if `properties.$active_feature_flags` is null or undefined.
+
+![Breakdown HogQL](../images/tutorials/array-filter-breakdown/breakdown.png)
+
+More useful array breakdowns include:
+
+- `arrayUniq` for counting the number of unique values in the array.
+- `arrayDistinct` which returns an array containing only the distinct elements. You can combine this with `arrayJoin` to get counts for distinct values within arrays.
+- `arrayElement(arr, 1)` for breaking down the first elements in the array.
+- `arrayElement(arr, -1)` for breaking down the last elements in the array.
+
+Once you get individual number or string values from the breakdown, you can modify these values further too. Read our [HogQL expression docs](/docs/hogql/expressions) for more useful functions for doing this.
+
+## Filtering arrays
+
+Once you break down your array, you might find it contains values you don't care about. You can use filters to remove them. Filtering arrays is relatively similar to breaking down, except you must provide values you want to filter out.
+
+To add a filter, click the filter dropdown next to your data series, click "Add filter," select HogQL, and add your expression.
+
+To start, you can remove empty arrays with a `notEmpty()` check. For example, to remove empty arrays from the `$active_feature_flags` by filtering for the HogQL expression:
+
+```sql
+notEmpty(
+  JSONExtractArrayRaw(
+    properties.$active_feature_flags ?? '[]'
+  )
+)
+```
+
+![Filter HogQL](../images/tutorials/array-filter-breakdown/filter.png)
+
+If you want to filter arrays for specific values, you can use `has`, `hasAll`, and `hasAny`:
+
+- `has()` filters for arrays containing an element. `has(arr, 'test')` matches arrays containing `'test'`.
+
+- `hasAll()` filters for arrays containing all of another array of elements. `hasAll(arr, ['test', 'cool'])` matches arrays containing both `'test'` and `'cool'`.
+
+- `hasAny()` filters for arrays containing any of another array of elements. `hasAny(arr, ['test', 'cool']` matches arrays containing either `'test'` or `'cool'`.
+
+If you want to filter arrays by the first, last, or specific index value, you can use `arrayElement(arr, n)` again too.
+
+## Further reading
+
+- [Using HogQL for advanced time and date filters](/tutorials/hogql-date-time-filters)
```

Comment:
This is separate from this specific article, but I think it's going to be worth creating a component that automatically populates tutorials here based on a tag. The number of tutorials is just going to keep growing and, as it does, keeping these lists updated will become more of a chore. A reusable component would be a good way to just keep things up to date. May be a W&D thing though.

Reply:
I haven't dug into it, but maybe. I don't think it is high priority though. Tags need to be fixed up first.